near-sdk/src/store/lookup_set/mod.rs

```diff
@@ -1,19 +1,40 @@
 mod impls;
 
 use crate::crypto_hash::{CryptoHasher, Sha256};
-use crate::store::LookupMap;
-use crate::IntoStorageKey;
+use crate::{env, IntoStorageKey};
 use borsh::{BorshDeserialize, BorshSerialize};
 use std::borrow::Borrow;
+use std::cell::{Cell, RefCell};
+use std::collections::BTreeMap;
 use std::fmt;
+use std::marker::PhantomData;
+
+const ERR_ELEMENT_SERIALIZATION: &str = "Cannot serialize element";
+
+type LookupKey = [u8; 32];
 
 #[derive(BorshSerialize, BorshDeserialize)]
 pub struct LookupSet<T, H = Sha256>
 where
     T: BorshSerialize + Ord,
     H: CryptoHasher<Digest = [u8; 32]>,
 {
-    map: LookupMap<T, (), H>,
+    prefix: Box<[u8]>,
+
+    /// Cache that keeps track the state of elements in the underlying set.
+    #[borsh_skip]
+    cache: RefCell<BTreeMap<T, Cell<EntryState>>>,
```

Comment:
I am not sure if wrapping `BTreeMap` in a `RefCell` here is a good idea, but I needed it to mutate the cache inside `contains`.

Reply:
Just use `StableMap` which is in our utils and it will give you what you need for this without doing this. It is this same API just that it allows for inserting with an immutable reference and still allows mutating with a mutable reference.

See how it's used in `LookupMap`

Reply:
The reason why I didn't use `StableMap` is because it wraps values in `Box` which I don't think I need, but let me give it another shot

Reply:
Ok, I think it indeed looks better now. PTAL